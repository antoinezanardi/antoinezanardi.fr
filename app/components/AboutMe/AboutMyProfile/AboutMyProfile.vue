<template>
  <div>
    <div class="card-body">
      <div class="h3 mt-0 title">
        <WrappedFontAwesomeIcon
          classes="me-2"
          icon="fa-user-circle"
        />

        <span>
          {{ $t('AboutMyProfile.whoAmI') }}
        </span>
      </div>

      <hr>

      <p v-html="$t('AboutMyProfile.welcomeOnMyProfile', { 'emoji': 'ðŸ‘‹' })"/>

      <p>
        <span v-html="$t('AboutMyProfile.myDiplomas')"/>

        <a
          class="mx-1"
          href="#education"
        >
          {{ $t('AboutMyProfile.myEducation') }}
        </a>

        <span v-html="$t('AboutMyProfile.bindWith')"/>

        <a
          class="mx-1"
          href="#experience"
          v-html="$t('AboutMyProfile.myExperience')"
        />

        <span v-html="$t('AboutMyProfile.giveMeManagementAndTechnicalSkills')"/>
      </p>

      <p v-html="`${$t('AboutMyProfile.myPassionAndPerfectionism')} ${$t('AboutMyProfile.myTechnologyWatch')}`"/>
    </div>
  </div>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";
</script>