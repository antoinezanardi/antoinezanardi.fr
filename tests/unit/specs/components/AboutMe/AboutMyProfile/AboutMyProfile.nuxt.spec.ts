import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

import AboutMyProfile from "@/components/AboutMe/AboutMyProfile/AboutMyProfile.vue";

describe("AboutMyProfile Component", () => {
  let wrapper: ReturnType<typeof mount<typeof AboutMyProfile>>;

  async function mountAboutMyProfileComponent(options: ComponentMountingOptions<typeof AboutMyProfile> = {}): Promise<ReturnType<typeof mount<typeof AboutMyProfile>>> {
    return mountSuspendedComponent(AboutMyProfile, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountAboutMyProfileComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});