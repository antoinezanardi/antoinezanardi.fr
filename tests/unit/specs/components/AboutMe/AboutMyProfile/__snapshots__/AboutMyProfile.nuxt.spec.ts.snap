// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutMyProfile Component > should match snapshot when rendered. 1`] = `
"<div>
  <div class="card-body">
    <div class="h3 mt-0 title">
      <wrapped-font-awesome-icon-stub icon="fa-user-circle" iconcolor="#000000" classes="me-2"></wrapped-font-awesome-icon-stub><span>Qui suis-je ?</span>
    </div>
    <hr>
    <p>Hello üëã ! Je suis <b>Antoine Zanardi</b> et bienvenue sur mon profil. Je suis <b> expert informatique sp√©cialis√© dans le web</b> et passionn√© par les nouvelles technologies, la robotique, l'intelligence artificielle et les innovations en tout genre.</p>
    <p><span>Mes dipl√¥mes acquis pendant</span><a class="mx-1" href="#education">mon cursus scolaire</a><span>coupl√©s √†</span><a class="mx-1" href="#experience">mon parcours professionnel</a><span>me permettent d'avoir un <b> esprit d'√©quipe </b> fort ainsi que des comp√©tences <b> techniques et manag√©riales</b> avanc√©es.</span></p>
    <p>Ma passion et mon perfectionnisme me poussent √† aller au bout de mes projets en y apportant <b>ma rigueur et mon savoir</b> pour les terminer dans les temps et dans<b> la qualit√© attendue</b>. Ma veille technologique me permet d'√™tre √† la page sur les nouveaut√©s du web et de les appliquer dans mes projets.</p>
  </div>
</div>"
`;
